### ISSUE TITLE
Implement a /ddo path for test

### ISSUE THREAD

**Issue by shark8me** [DEX-Company/koi-clj#55]
Notes from an email:

```sh
I suggest implementing a "connect" starfish feature that creates and installs a ddo for any target agent.

Potential features:
- Takes a URL of the agent as an argument
- Polls some well-defined path e.g. "/ddo" to get a DDO from the agent if available, which should include a globally known DID
- As a fallback approach, generates a local ddo that references default paths to the host endpoints e.g. URL+"/api/v1/invoke"

This could be used for testing purposes, but it also fulfils the need of being able to dynamically connect to an agent in the absence of a working resolver. Admiral will need this.

Some other considerations:
- Presumably needs to support provision of credentials as a map to enable the connection to be authenticated
```

**Comment by mikera** [comment]
@superstructor good if you could comment, one of the key uses for this is to support tools like Admiral!

**Comment by shark8me** [comment]
@mikera There is a dependency problem here.

- Surfer can start independently without any dependencies, and serve up a DDO.
- Koi needs a Surfer (Meta + Storage endpoint) to store meta+data for all assets generated by the operation(s) hosted by it.
- In that case, a tester using Koi would need to call `setMetaEndpoint` / `addMetaEndpoint` / `addStorageEndpoint`, before calling an Operation on Koi. 
- Koi would also need a resolver in order to get Asset data + metadata for assets that are passed in as arguments to Operations, if the list of Meta endpoints does not include the (Asset) DID.

**Comment by mikera** [comment]
Koi should only report it's own endpoints in its own DDO (presumably including the meta endpoints for its own hosted operations, and the storage endpoint if it also is configured to support upload/download of inputs/results)

If it is configured to use a surfer instance to store result assets, then it communicates that back to the caller with the appropriate DIDs in the result. The caller will of course need to be able to resolve these.

**Comment by shark8me** [comment]
Implemented in PR #60
